tools/shared.py

```diff
@@ -631,11 +631,11 @@ def build_clang_tool_path(tool):
 def macos_find_native_sdk_path():
   try:
     sdk_root = '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs'
-    sdks = os.walk(sdk_root).next()[1]
+    sdks = next(os.walk(sdk_root))[1]
     sdk_path = os.path.join(sdk_root, sdks[0]) # Just pick first one found, we don't care which one we found.
     logger.debug('Targeting macOS SDK found at ' + sdk_path)
     return sdk_path
-  except OSError:
+  except (OSError, StopIteration):
```

Comment:
This is kind of gross .. not sure why we need a try/catch here all there.

Reply:
`StopIteration` is raised when the iterator is exhausted, and the code was using `except:` to implicitly catch the "exception" to handle the case of the directory not existing.

Reply:
I would rather just use `if os.path.isdir(xxx)`, but that be a followup if you prefer.

Reply:
Technically, that's worse because of TOCTTOU. But I agree the code can use some refactoring. What it really needs is `os.listdir` and catching the resulting `OSError`, not this `os.walk` and `next` insanity. I'd rather just get the CI fixed first.

Reply:
That might be true in some cases but I don't think its true here.

Anyone who is running emcc while racing with installing or deleting the mac SDK directory on their system is going to get an exception and backtrace at some point either way.   Trying to handle that case here by catching here doesn't really buy us anything IMHO.